src/Components/TaskItemInput/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { ChangeEvent, useState, useEffect, useRef } from 'react';
+import { Input } from './TaskItemInput';
+
+export interface Props {
+    value: string;
+    dataCy?: string;
+    placeholder?: string;
+    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+    onEscape: () => void;
+}
+
+function TaskItemInput(props: Props) {
+
+    const [value, setValue] = useState(props.value);
+    const ref = useRef<any>(null);
+
+    useEffect(() => {
+        const onEscape = (event: KeyboardEvent) => {
+            event.stopPropagation();
+            if (event.key !== "Escape") return; 
+            setValue(props.value);
+            props.onEscape();
+        }
+
+        setValue(props.value);
+
+        document.addEventListener("keydown", onEscape, false);
+        return () => document.removeEventListener("keydown", onEscape, false);
+    }, [props])
```

Comment:
dx: this could easily become a custom hook. it would also increase code readability.